frame/contracts/src/lib.rs

```diff
@@ -372,6 +380,59 @@ pub mod pallet {
 				T::WeightInfo::on_process_deletion_queue_batch()
 			}
 		}
+
+		fn integrity_test() {
+			// Total runtime memory is expected to have 1Gb upper limit
+			const MAX_RUNTIME_MEM: u32 = 1024 * 1024 * 1024;
+			// Memory limits for a single contract:
+			// Value stack size: 1Mb per contract, default defined in wasmi
+			const STACK_MAX_SIZE: u32 = 1024 * 1024;
+			// Heap limit is normally 16 mempages of 64kb each = 1Mb per contract
+			let heap_max_size = T::Schedule::get().limits.max_memory_size();
```

Comment:
Sounds a bit like joda :D

```suggestion
			let max_heap_size = T::Schedule::get().limits.max_memory_size();
```